framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -763,6 +763,10 @@ private void logCheckerError(CheckerError ce) {
             if (this.currentRoot != null && this.currentRoot.getSourceFile() != null) {
                 msg.append("\nCompilation unit: " + this.currentRoot.getSourceFile().getName());
             }
+            if (this.visitor != null) {
+                Tree lv = this.visitor.lastVisited;
+                msg.append("\nLast visited tree: " + lv);
```

Comment:
Consider including the line and column position in the diagnostic, and printing the source line corresponding to the tree's diagnostic position ([example](https://github.com/google/error-prone/blob/66c0e96b90a8fb28a2d3febde5bef3b339b9f6cc/check_api/src/main/java/com/google/errorprone/ErrorProneError.java#L61)).

Prettying-printing the entire tree will result in a lot of diagnostic output if the last tree was e.g. a class or compilation unit, and it might not uniquely the position of the crash for small subexpressions.

Reply:
I agree and changed to line/col information. Thanks.